src/main/java/me/eigenraven/personalspace/world/DimensionConfig.java

```diff
@@ -198,11 +213,17 @@ public int syncWithFile(File file, boolean write, int dimId) {
         } else {
             setWeatherEnabled(cur.getBoolean());
         }
-        cur = cfg.get(VISUAL, "nightTime", nightTime, "");
+        cur = cfg.get(VISUAL, "daylightCycle", daylightCycle.ordinal(), "");
         if (write) {
-            cur.set(nightTime);
+            cur.set(daylightCycle.ordinal());
         } else {
-            setNightTime(cur.getBoolean());
+            setDaylightCycle(DaylightCycle.fromOrdinal(cur.getInt()));
+        }
+        // handle old nightTime config
+        if (!write && cfg.hasKey(VISUAL, "nightTime")) {
+            Boolean isNight = cfg.getCategory(VISUAL).get("nightTime").getBoolean();
+            setDaylightCycle(isNight ? DaylightCycle.MOON : DaylightCycle.SUN);
+            cfg.getCategory(VISUAL).remove("nightTime");
```

Comment:
You need to set `needsSaving` to true after removing it otherwise the config might never get updated

Reply:
`needsSaving` is set to false at the end of hat method

Reply:
ah it should probably be in the if(write) block, it didn't really make a difference before

Reply:
so like that?

Reply:
No, I meant to move the `needsSaving=false;` line at the end of the function inside the if(write) block just above it. The old format needs to be parsed on read, or you'll break old nighttime worlds

Reply:
Got it